### PR TITLE
UI: deck: order lists correctly

### PR DESCRIPTION
**Pull request by AcipenserSturio**

### CODE REVIEW

File: zzre/game/systems/ui/ScrDeck.cs [hunk 1]
```diff
@@ -443,10 +443,10 @@ private void CreateGridList(DefaultEcs.Entity entity, ref components.ui.ScrDeck
 
     private IEnumerable<InventoryCard> AllCardsOfType(in components.ui.ScrDeck deck) => deck.ActiveTab switch
     {
-        Tab.Items => deck.Inventory.Items,
-        Tab.Fairies => deck.Inventory.Fairies,
-        Tab.AttackSpells => deck.Inventory.AttackSpells,
-        Tab.SupportSpells => deck.Inventory.SupportSpells,
+        Tab.Items => deck.Inventory.Items.OrderBy(c => c.cardId.EntityId),
+        Tab.Fairies => deck.Inventory.Fairies.OrderBy(c => -c.level),
```
Comment (Helco) [inline zzre/game/systems/ui/ScrDeck.cs:447]: nitpick: [`OrderByDescending`](https://learn.microsoft.com/de-de/dotnet/api/system.linq.enumerable.orderbydescending?view=net-8.0) would be slightly more clear

File: zzre/game/systems/ui/ScrDeck.cs [hunk 2]
```diff
@@ -443,10 +443,10 @@ private void CreateGridList(DefaultEcs.Entity entity, ref components.ui.ScrDeck
 
     private IEnumerable<InventoryCard> AllCardsOfType(in components.ui.ScrDeck deck) => deck.ActiveTab switch
     {
-        Tab.Items => deck.Inventory.Items,
-        Tab.Fairies => deck.Inventory.Fairies,
-        Tab.AttackSpells => deck.Inventory.AttackSpells,
-        Tab.SupportSpells => deck.Inventory.SupportSpells,
+        Tab.Items => deck.Inventory.Items.OrderBy(c => c.cardId.EntityId),
```
Comment (Helco) [inline zzre/game/systems/ui/ScrDeck.cs:446]: There the sorting is unfortunately not quite correct, but Zanzarah is a bit weird with this:
There is a column in the ItemRow, currently still called `Unknown` (should probably be `Type`) that is used as first sorting term with the order `1 -> 0 -> everything else`
The `EntityId` is used as a secondary sorting term

**Review by Helco — CHANGES_REQUESTED** [review]

File: zzre/game/systems/ui/ScrDeck.cs (hunk 1 above)
Comment (AcipenserSturio) [reply zzre/game/systems/ui/ScrDeck.cs]: Glad to know more api, fixed

File: zzre/game/systems/ui/ScrDeck.cs (hunk 2 above)
Comment (AcipenserSturio) [reply zzre/game/systems/ui/ScrDeck.cs]: fixed
that's so obscure lol

**Review by Helco — APPROVED** [review]
Great stuff!

**Pull request merged**